libcaf_core/caf/flow/observable.test.cpp

```diff
@@ -438,50 +421,46 @@ TEST("on_error_return() replaces an error with a value") {
   SECTION("on_error_return() replaces an error with a value from the handler") {
     auto return_42 = [](const error&) { return expected<int>{42}; };
     SECTION("blueprint") {
-      check_eq(collect(range(1, 0)
-                         .concat(obs_error(sec::runtime_error))
-                         .on_error_return(return_42)),
+      check_eq(collect(
+                 range(1, 0).concat(obs_error()).on_error_return(return_42)),
                vector{42});
       check_eq(collect(range(1, 2)
-                         .concat(obs_error(sec::runtime_error))
+                         .concat(obs_error())
                          .concat(range(1, 2))
                          .on_error_return(return_42)),
                vector{1, 2, 42});
       check_eq(collect(range(1, 3)
-                         .concat(obs_error(sec::runtime_error))
+                         .concat(obs_error())
                          .concat(range(1, 3))
                          .on_error_return(return_42)),
                vector{1, 2, 3, 42});
-      check_eq(collect(range(1, 2)
-                         .concat(obs_error(sec::runtime_error))
-                         .on_error_return(return_42)
-                         .take(2)),
-               vector{1, 2});
-      check_eq(collect(range(1, 3)
-                         .concat(obs_error(sec::runtime_error))
-                         .on_error_return(return_42)
-                         .take(3)),
-               vector{1, 2, 3});
+      check_eq(
+        collect(
+          range(1, 2).concat(obs_error()).on_error_return(return_42).take(2)),
+        vector{1, 2});
+      check_eq(
+        collect(
+          range(1, 3).concat(obs_error()).on_error_return(return_42).take(3)),
+        vector{1, 2, 3});
     }
     SECTION("observable") {
-      check_eq(collect(mat(range(1, 0).concat(obs_error(sec::runtime_error)))
-                         .on_error_return(return_42)),
-               vector{42});
-      check_eq(collect(mat(range(1, 2)
-                             .concat(obs_error(sec::runtime_error))
-                             .concat(range(1, 2)))
-                         .on_error_return(return_42)),
+      check_eq(
+        collect(
+          build(range(1, 0).concat(obs_error())).on_error_return(return_42)),
```

Comment:
Similarly this can be refactored to
```suggestion
          check_eq(collect(build(obs_error().on_error_return(return_42))), vector{42});
```